tests/schema_test.py

```diff
@@ -6,37 +6,53 @@
 schema.
 
 """
+import collections
+import unittest
 
 from .utils import BaseTest, check_for_cycles, validate_schemata
 
 
 class SchemaTest(BaseTest):
 
-    def traverse_schema_with_conditional(self, root, conditional_func, test_func):
-        if conditional_func(root):
-            test_func(root)
-
-        if isinstance(root, list):
-            for child in root:
-                self.traverse_schema_with_conditional(child, conditional_func, test_func)
-        elif isinstance(root, dict):
-            for child in root.keys():
-                self.traverse_schema_with_conditional(root[child], conditional_func, test_func)
-
-    def test_enum_is_string(self):
-
-        def trigger(root):
-            if isinstance(root, dict) and 'enum' in root.keys():
-                return True
-            return False
-
-        def test_enum(root):
-            children = root['enum']
-            assert isinstance(children, list), "Enums children wasn't a list"
-            for child in children:
-                assert isinstance(child, str), "Offending enum found:" + str(child)
-
-        self.traverse_schema_with_conditional(self.dictionary.schema, trigger, test_enum)
+    @unittest.expectedFailure
+    def test_properties_enum(self):
+        """Check the enums of node properties"""
+        # The enums in _definitions.yaml, _terms.yaml and metaschema.yaml are not checked
+
+        def check_enum(enums):
```

Comment:
This reads a lot better. I still thing that this function and the one below are better suited as methods in the test class itself.

Reply:
It seems that we prefer module level functions over static methods and class methods, so I moved them outside of the class (ref: https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators)